src/ctapipe/monitoring/interpolation.py

```diff
@@ -184,5 +209,139 @@ def add_table(self, tel_id, input_table):
         alt = input_table["altitude"].quantity.to_value(u.rad)
         mjd = input_table["time"].tai.mjd
         self._interpolators[tel_id] = {}
-        self._interpolators[tel_id]["az"] = interp1d(mjd, az, **self.interp_options)
-        self._interpolators[tel_id]["alt"] = interp1d(mjd, alt, **self.interp_options)
+        self._interpolators[tel_id]["az"] = interp1d(
+            mjd, az, kind="linear", **self.interp_options
+        )
+        self._interpolators[tel_id]["alt"] = interp1d(
+            mjd, alt, kind="linear", **self.interp_options
+        )
+
+
+class ChunkInterpolator(MonitoringInterpolator):
+    """
+    Simple interpolator for overlapping chunks of data.
+    """
+
+    def __init__(self, h5file: None | tables.File = None, **kwargs: Any) -> None:
+        super().__init__(**kwargs)
+        self._interpolators = {}
+        self.required_columns = ["start_time", "end_time"]
```

Comment:
required columns as start and stop time shall be the class attributes and shall be frozen, they are *mandatory*. You can copy them to an instance and extend with a value column(s) when you engage a `__call__`.

Reply:
as the functions that use required_columns are in the parent class and always look to that name it makes more sense to have it as a modifiable instance variable

Reply:
No. The original design idea was to have the required_columns frozen per "final" class. I.e. `MonitoringInterpolator` requires `altitude` and `azimuth` columns.

This is due to the configuration system in ctapipe, which works on class name basis, not instances.